aws: documentation fixes

PettitWesley opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **Documentation**
  * Improved clarity of AWS output plugin documentation with refined instructions and wording for configuration examples.
  * Enhanced guidance for handling public ECR image pull limit errors and AWS credentials.
  * Improved formatting and organization of technical sections for better readability.

<!-- review_stack_entry_start -->

[![Review Change Stack](https://storage.googleapis.com/coderabbit_public_assets/review-stack-in-coderabbit-ui.svg)](https://app.coderabbit.ai/change-stack/fluent/fluent-bit-docs/pull/880?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

<!-- review_stack_entry_end -->

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

PettitWesley committed:
out_cloudwatch_logs: fix doc language and formatting

Signed-off-by: Wesley Pettit <wppttt@amazon.com>

PettitWesley committed:
out_kinesis_firehose: fix doc language and formatting

Signed-off-by: Wesley Pettit <wppttt@amazon.com>

PettitWesley committed:
out_kinesis_streams: fix doc language and formatting

Signed-off-by: Wesley Pettit <wppttt@amazon.com>

PettitWesley committed:
out_s3: fix doc language and formatting

Signed-off-by: Wesley Pettit <wppttt@amazon.com>

DrewZhang13 reviewed (approved):
LGTM

lecaros commented on pipeline/outputs/kinesis.md:
```diff
@@ -8,7 +8,7 @@ description: Send logs to Amazon Kinesis Streams
 The Amazon Kinesis Data Streams output plugin allows to ingest your records into the [Kinesis](https://aws.amazon.com/kinesis/data-streams/) service.
 
-This is the documentation for the core Fluent Bit Kinesis plugin written in C. It has all the core features of the [aws/amazon-kinesis-streams-for-fluent-bit](https://github.com/aws/amazon-kinesis-streams-for-fluent-bit) Golang Fluent Bit plugin released in 2019. The Golang plugin was named `kinesis`; this new high performance and highly efficient kinesis plugin is called `kinesis_streams` to pre ...
+This is the documentation for the core Fluent Bit Kinesis plugin written in C. It has all the core features of the [aws/amazon-kinesis-streams-for-fluent-bit](https://github.com/aws/amazon-kinesis-streams-for-fluent-bit) Golang Fluent Bit plugin released in 2019. The Golang plugin was named `kinesis`; this new high performance and highly fficient kinesis plugin is called `kinesis_streams` to prev ...
```
Typo: efficient was changed to fficient (missing e)

lecaros reviewed (changes requested):
Please review the typo

matthewfala reviewed (changes requested):
They look great! Just one typo maybe

matthewfala commented on pipeline/outputs/kinesis.md:
```diff
@@ -8,7 +8,7 @@ description: Send logs to Amazon Kinesis Streams
 The Amazon Kinesis Data Streams output plugin allows to ingest your records into the [Kinesis](https://aws.amazon.com/kinesis/data-streams/) service.
 
-This is the documentation for the core Fluent Bit Kinesis plugin written in C. It has all the core features of the [aws/amazon-kinesis-streams-for-fluent-bit](https://github.com/aws/amazon-kinesis-streams-for-fluent-bit) Golang Fluent Bit plugin released in 2019. The Golang plugin was named `kinesis`; this new high performance and highly efficient kinesis plugin is called `kinesis_streams` to pre ...
+This is the documentation for the core Fluent Bit Kinesis plugin written in C. It has all the core features of the [aws/amazon-kinesis-streams-for-fluent-bit](https://github.com/aws/amazon-kinesis-streams-for-fluent-bit) Golang Fluent Bit plugin released in 2019. The Golang plugin was named `kinesis`; this new high performance and highly fficient kinesis plugin is called `kinesis_streams` to prev ...
```
Just saw Lecaros mentioned the same thing

PettitWesley commented on pipeline/outputs/cloudwatch.md:
```diff
@@ -8,46 +8,46 @@ description: Send logs and metrics to Amazon CloudWatch
+| `log_format`          | An optional parameter that can be used to tell CloudWatch the format of the data. A value of `json/emf` enables CloudWatch to extract custom metrics embedded in a JSON payload. See the [Embedded Metric Format](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch_Embedded_Metric_Format_Specification.html).                                               ...
+| `role_arn`            | The value of this option must be ARN of an IAM role. This role will be assumed using an `sts:AssumeRole` call. This parameter is typically used to send logs from one AWS account to another.                                                                                                                                                                                         ...
+| `auto_create_group`   | Automatically create the log group. Valid values are `true`/`on` and `false`/`off` (case insensitive). Defaults to `false`.                                                                                                                                                                                                                                                           ...
+| `log_retention_day`s  | If set to a number greater than zero, and newly create log group's retention policy is set to this many days. Valid values are: \[1, 3, 5, 7, 14, 30, 60, 90, 120, 150, 180, 365, 400, 545, 731, 1827, 3653]                                                                                                                                                                          ...
```
oops

eschabell commented:
@PettitWesley can you resolve the conflicts and I'll get this PR merged for you?

eschabell commented:
@esmerel can you take a look at this PR?

esmerel commented:
This one is tougher due to the interweaving of a couple of things in the cloudwatch page, I'll have to come back to this one.

eschabell committed:
Merge branch 'master' into sandra-doc-fixes

Signed-off-by: Eric D. Schabell <eric@schabell.org>

eschabell committed:
docs: pipeline: outputs: cloudwatch: fix broken LocalStack code block

  - Restore opening ```yaml fence and pipeline: key removed in a prior edit,
    which left the LocalStack YAML example rendering as plain text
  - Remove informal 'Recall that...' sentence inserted before Kubernetes
    record_accessor example

    Applies to #880

Signed-off-by: Eric D. Schabell <eric@schabell.org>

eschabell commented on pipeline/outputs/cloudwatch.md:
```diff
@@ -193,6 +192,8 @@ Here is the configuration:
 {% tab title="fluent-bit.yaml" %}
 
+Recall that the Kubernetes filter can add metadata which will look like the following:
+
```
Fixed.

eschabell reviewed (approved):
@PettitWesley thanks for the docs PR, merging.
